### PR TITLE
Set distributor.remote-timeout=10s for rulers in jsonnet/helm

### PR DESCRIPTION
**Pull request by leizor**
#### What this PR does

Internally, we had adjusted the value of `distributor.remote-timeout` for rulers from 2s to 10s in order to accommodate writing large rule results to the ingester. This PR upstreams those changes.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by 56quarters — CHANGES_REQUESTED** [review]
The internal change you mention was _just_ for rulers writing to ingesters unless I missed something. I disagree that `10s` should be the new default for the CLI option, jsonnet, or helm chart. That's way too long a timeout for an RPC request.

**Comment by leizor** [comment]
Oops, I misunderstood - I'll get it fixed.

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 ## main / unreleased
 
 * [CHANGE] Rollout-operator: remove default CPU limit. #7125
+* [CHANGE] Ruler: Set `distributor.remote-timeout` to 10s in order to accommodate writing large rule results to the ingester. #7143
```
Comment (pracucci) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:32]: [nit] We're used to prefix CLI flags with `-` to clarify it's a CLI flag.

```suggestion
* [CHANGE] Ruler: Set `-distributor.remote-timeout` to 10s in order to accommodate writing large rule results to the ingester. #7143
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 2]
```diff
@@ -1097,7 +1097,8 @@ ruler:
       memory: 128Mi
 
   # Additional ruler container arguments, e.g. log level (debug, info, warn, error)
-  extraArgs: {}
+  extraArgs:
+    distributor.remote-timeout: 10s
```
Comment (pracucci) [inline operations/helm/charts/mimir-distributed/values.yaml:1101]: I've been told by @krajorama in other PR to not keep `extraArgs` empty and just set it right after `- "-config.file=/etc/mimir/mimir.yaml"` in `operations/helm/charts/mimir-distributed/templates/ruler/ruler-dep.yaml`. If an user wants to override it, then setting it in `extraArgs` will take precedence.

File: operations/mimir/ruler.libsonnet [hunk 3]
```diff
@@ -30,6 +30,8 @@
       'ruler.ring.consul.hostname': 'consul.%(namespace)s.svc.%(cluster_domain)s:8500' % $._config,
 
       'server.http-listen-port': $._config.server_http_port,
+
+      'distributor.remote-timeout': '10s',
```
Comment (pracucci) [inline operations/mimir/ruler.libsonnet:34]: [nit] I suggest to add a comment to explain "why" (basically what you say in the PR description: "... to accommodate writing large rule results to the ingester").

**Review by pracucci — COMMENTED** [review]
Thanks! I left a comment about Helm.

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 2 above)
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml:1101]: i agree. it also means this change will take effect if the user already has some other parameters configured in `extraArgs`

Comment (leizor) [reply operations/helm/charts/mimir-distributed/values.yaml:1101]: Good to know, thanks for the tip!

Just for my own edification - I'm a little unclear on how how the override would work. If the user adds the `-distributor.remote-timeout` flag to `extraArgs`, wouldn't that flag then appear in the ruler's args list twice?

Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml]: > wouldn't that flag then appear in the ruler's args list twice?

that's right. Go's default behaviour is to take the value from the last usage of the flag. I couldn't find docs, but you can use this go program to prove verify it experimentally (by running all possible values of the flag in all possible orders? idk) https://go.dev/play/p/Qq2uqAHiFuQ


```go
package main

import (
	"flag"
	"fmt"
)

func main() {
	var f string
	flag.StringVar(&f, "flag", "", "help message for flagname")
	flag.Parse()
	fmt.Println("f:", f)
}
```

```
go run main.go -flag=123 -flag=qwerty
f: qwerty
```

Comment (leizor) [reply operations/helm/charts/mimir-distributed/values.yaml]: Gotcha - thanks Dimitar!

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM

**Review by pracucci — APPROVED** [review]
Thanks for addressing my feedback, LGTM!

**Pull request merged**